src/sagas/storageDomains.js

```diff
@@ -20,111 +20,119 @@ import {
+    const dataCentersInternal = dataCentersApi.data_center.map(
+      dataCenter => Transforms.DataCenter.toInternal({ dataCenter })
+    )
+
+    // Calculate permits and 'canUser*'
+    for (const dataCenter of dataCentersInternal) {
+      dataCenter.userPermits = yield permissionsToUserPermits(dataCenter.permissions)
     }
+
+    return dataCentersInternal
   }
 
-  // since storage domains can be attached to >1 data center, collate the copies
-  // and track the domain's status per data center
-  const storageDomainsMerged = mergeStorageDomains(storageDomainsInternal)
-  yield put(setStorageDomains(storageDomainsMerged))
+  return []
 }
 
-/**
- * From a list of storage domains, where a storage domain may occur multiple times
- * with a different data center, build a unique list of storage domains.  Each storage
- * domain returned will contain a data center to SD status map.
- */
-function mergeStorageDomains (storageDomainsInternal) {
-  const idToStorageDomain = storageDomainsInternal.reduce(
-    (accum, storageDomain) => {
-      const merged = accum[storageDomain.id]
-      if (merged) {
-        Object.assign(merged.statusPerDataCenter, storageDomain.statusPerDataCenter)
-      } else {
-        accum[storageDomain.id] = storageDomain
-      }
-      return accum
-    },
-    {}
-  )
+function* fetchDataAndIsoStorageDomains () {
+  const storageDomainsApi = yield callExternalAction('getStorages', Api.getStorages, { payload: {} })
+
+  if (storageDomainsApi && storageDomainsApi.storage_domain) {
+    const storageDomainsInternal = storageDomainsApi.storage_domain
+      .map(storageDomain => Transforms.StorageDomain.toInternal({ storageDomain }))
+      .filter(storageDomain => storageDomain.type === 'data' || storageDomain.type === 'iso')
+
+    // Calculate permits and 'canUser*'
+    for (const storageDomain of storageDomainsInternal) {
+      storageDomain.userPermits = yield permissionsToUserPermits(storageDomain.permissions)
+      storageDomain.canUserUseDomain = canUserUseStorageDomain(storageDomain.userPermits)
+    }
+
+    return storageDomainsInternal
+  }
 
-  const mergedStorageDomains = Object.values(idToStorageDomain)
-  return mergedStorageDomains
+  return []
 }
 
 /**
  * Fetch all of the ISO images from both 'iso' type storage domains and 'iso' types
  * images from other types of storage domains.
  */
 export function* fetchIsoFiles (action) {
-  // fetch ISO disk images and distribute them to their storage domains as files
-  const images = yield callExternalAction('getIsoImages', Api.getIsoImages, { payload: {} })
-  if (images && images.disk) {
-    const storageDomainToDisks = images.disk.reduce(
-      (acc, disk) => {
-        disk.storage_domains.storage_domain.forEach(({ id }) => {
-          const files = acc[id] = acc[id] || []
-          files.push(Api.fileToInternal({ file: disk }))
-        })
-        return acc
-      },
-      {}
-    )
+  yield all([
+    call(function* () {
```

Comment:
I think it would be nice to move this function outside, because it is not very clear purpose of this function.